osf/migrations/0177_update_schema_data.py

```diff
@@ -0,0 +1,285 @@
+    Some original schemas have html in the description/help text.  Just replacing this with empty strings.
+    """
+    stripped_html = re.sub('<.*?>', '', string_with_html)
+    return stripped_html
+
+def find_title_description_help_example(rs, question):
+    """
+    Schemas are inconsistent with regards to the information going into "title",
+    "description", and "help" blocks.
+
+    :returns tuple, title, description, help, example strings
+
+    """
+    title = question.get('title', '')
+    description = strip_html(question.get('description', ''))
+    help = strip_html(question.get('help', ''))
+    example = ''
+
+    schema_name = rs.schema.get('name', '')
+    # Descriptions that contain any of these keywords
+    # are turned into help text instead.
+    help_text_keywords = [
+        'please',
+        'choose',
+        'provide',
+        'format',
+        'describe',
+        'who',
+        'what',
+        'when',
+        'where',
+        'use',
+        'you',
+        'your',
+        'skip',
+        'enter',
+    ]
+
+    if title:
+        if schema_name in ['OSF Preregistration', 'Prereg Challenge']:
+            # These two schemas have clear "example" text in the "help" section
+            example = help
+            help = description
+            description = ''
+        else:
+            for keyword in help_text_keywords:
+                if keyword in description.lower():
+                    help = description
+                    description = ''
+                    break
+    else:
+        # if no title, description text is moved to title.
+        title = description
+        description = ''
+
+    return title, description, help, example
+
+def format_property(question, property, index):
+    """
+    Modify a subquestion's qid, title, description, and help text, because
+    these are often absent.
+    - Modify a subquestion's qid to be of the format "parent-id.current-id", to
+      reflect its nested nature, to ensure uniqueness
+    - For the first nested subquestion, transfer the parent's title, description, and help.
+    """
+    property['qid'] = '{}.{}'.format(get_answer_id(question), property.get('id', ''))
+    if not index:
+        title = question.get('title', '')
+        description = question.get('description', '')
+        help = question.get('help', '')
+        if not property.get('title', '') and not property.get('description'):
+            property['title'] = title
+        if not property.get('description', ''):
+            property['description'] = description
+        if not property.get('help', ''):
+            property['help'] = help
+    return property
+
+
+def format_question(state, rs, question, sub=False):
+    """
+    Split the original question from the schema into multiple schema blocks, all of
+    which have the same chunk_id, to link them.
+    """
+    # If there are subquestions, recurse and format subquestions
+    if question.get('properties'):
```

Comment:
if `question` has a `properties` key - then I'm making a `section-heading` or a `subsection-heading`, depending.  Some `section-headings` only have one nested question, but many have two (like a free text and then a file-upload).  Did the logic this way, for consistency.  

Veer `"Pre-Registration in Social Psychology (van 't Veer & Giner-Sorolla, 2016): Pre-Registration` is the only schema which has `subsection-headings` - three levels of nesting.